UnityProject/Assets/Scripts/Items/Weapons/Melee/MeleeEffect.cs

```diff
@@ -118,14 +118,22 @@ public void ServerPerformInteraction(HandApply interaction)
 			Vector2 dir = (target.transform.position - performer.transform.position).normalized;
 
 			WeaponNetworkActions wna = performer.GetComponent<WeaponNetworkActions>();
+			ToggleableEffect toggleableEffect = gameObject.GetComponent<ToggleableEffect>();
 
 			// If we're on harm intent we deal damage!
 			// Note: this has to be done before the teleport, otherwise the target may be moved out of range.
-			if (interaction.Intent == Intent.Harm)
+			if (interaction.Intent == Intent.Harm && toggleableEffect.CurrentWeaponState == ToggleableEffect.WeaponState.On)
```

Comment:
One problem, when toggleableEffect.CurrentWeaponState != ToggleableEffect.WeaponState.On, the meleeEffect component will actually be disabled.  So the feedback message won't show.

Reply:
but it showed on my end?

Reply:
It showed when the baton was turned off?

Reply:
oh i get what you mean now.
I didn't notice that the component gets disabled when the baton is off, will find a way to keep it working.

Reply:
Fixed